### ISSUE TITLE
Get Strong Typed Metric From paddle::Evaluator

### ISSUE THREAD

**Issue by reyoung** [PaddlePaddle/Paddle#1389]
In current Paddle API in Python, we should get the strong-typed metric when training, such as Error Rate in float. 

We should add some interfaces to C++ class `paddle::Evaluator` to get the metric result firstly, and then we could expose them to SWIG/Python.

Each current evaluator could contain zero to many metrics inside. 

* each evaluator should include one metric result, like classification evaluator, contains ErrorRate.
* There are some evaluators just used for debugging, they often called XXXPrinter.
* There are some evaluators will contain many metric results, like precision-recall evaluator. It because some metrics are calculated together.

So basically, the metrics in Evaluator should be a map or dictionary, the key is a metric name, the value is the strong-typed result.
